samples/nrf9160/gnss/src/main.c

```diff
@@ -278,21 +269,67 @@ static int gnss_init_and_start(void)
 		return -1;
 	}
 
-	if (nrf_modem_gnss_nmea_mask_set(NRF_MODEM_GNSS_NMEA_RMC_MASK |
-					 NRF_MODEM_GNSS_NMEA_GGA_MASK |
-					 NRF_MODEM_GNSS_NMEA_GLL_MASK |
-					 NRF_MODEM_GNSS_NMEA_GSA_MASK |
-					 NRF_MODEM_GNSS_NMEA_GSV_MASK) != 0) {
+	/* Enable all NMEAs. */
+	uint16_t nmea_mask = NRF_MODEM_GNSS_NMEA_RMC_MASK |
+			     NRF_MODEM_GNSS_NMEA_GGA_MASK |
+			     NRF_MODEM_GNSS_NMEA_GLL_MASK |
+			     NRF_MODEM_GNSS_NMEA_GSA_MASK |
+			     NRF_MODEM_GNSS_NMEA_GSV_MASK;
+
+	if (nrf_modem_gnss_nmea_mask_set(nmea_mask) != 0) {
 		LOG_ERR("Failed to set GNSS NMEA mask");
 		return -1;
 	}
 
-	if (nrf_modem_gnss_fix_retry_set(0) != 0) {
+	/* This use case flag should always be set. */
+	uint8_t use_case = NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START;
+
+#if defined(CONFIG_GNSS_SAMPLE_MODE_PERIODIC) && \
+	!defined(CONFIG_GNSS_SAMPLE_ASSISTANCE_NONE)
```

Comment:
Nit: You could align like this to avoid it looking like `!defined..` is part of the code block
```
#if defined(CONFIG_GNSS_SAMPLE_MODE_PERIODIC) && \
	    !defined(CONFIG_GNSS_SAMPLE_ASSISTANCE_NONE)
```

Reply:
Generally, I'd consider replacing #ifdefs with C code and IS_ENABLED() macro here and below.

Reply:
I'll update the code to use IS_ENABLED() macro instead.